src/libsyntax/parse/parser.rs

```diff
@@ -4381,9 +4381,14 @@ impl<'a> Parser<'a> {
             if let Ok(seq_snippet) = self.sess.source_map().span_to_snippet(seq_span) {
                 err.span_suggestion(
                     seq_span,
-                    "try adding parentheses",
+                    "try adding parentheses..",
                     format!("({})", seq_snippet),
                     Applicability::MachineApplicable
+                ).span_suggestion(
+                    seq_span,
+                    "..or a vertical bar",
+                    format!("{}", seq_snippet.replace(",", " | ")),
```

Comment:
```suggestion
                    format!("{}", seq_snippet.replace(",", " |")),
```

This adds one additional space that already correctly formatted code would not need